src/extensions/default/UploadFiles/UploadFilesDialog.js

```diff
@@ -164,13 +166,16 @@ define(function (require, exports, module) {
         Dialogs.cancelModalDialogIfOpen("upload-files-dialog");
     };
     FileUploadDialog.prototype.destroy = function() {
+        _uploadDialog = null;
         this.fileInput.remove();
     };
 
-
     function show() {
-        var uploadDialog = new FileUploadDialog();
-        return uploadDialog.show();
+	if(_uploadDialog) {
+	    return _uploadDialog;
```

Comment:
I feel like we should be returning `_uploadDialog.deferred.promise()` here, but I'm not sure. What do you think @humphd?